glean-core/android/src/main/java/mozilla/telemetry/glean/private/BooleanMetricType.kt

```diff
@@ -32,6 +33,13 @@ class BooleanMetricType(
                 disabled = disabled.toByte())
     }
 
+    protected fun finalize() {
```

Comment:
@badboy, is this a Rust specific behaviour/trait? When does this get called?

Reply:
Oh wait, this is Kotlin code. So it must be what Mike mentioned [here](https://kotlinlang.org/docs/reference/java-interop.html#finalize). But that should be for imported Java object: did you check this works?

Reply:
Kinda. I added a print statement and run the tests and checked the stdout log to contain the finalize thing, so yeah, it gets called (and didn't crash).

Of course finalizers are not guaranteed to run (or its at least nondeterministic _when_ they run).